bossbot/db.py

```diff
@@ -27,11 +27,38 @@ def __init__(
             cursor.execute(GR_SETUP_TABLE)
             cursor.execute(GR_OAUTH_TABLE)
             cursor.execute(LASTFM_USER_TABLE)
+            cursor.execute(KARMA_TRACKER_TABLE)
             self.conn.commit()
 
     def cursor(self, *args, **kwargs):
         return self.conn.cursor(*args, **kwargs)
 
+class KarmaDB(BossDB):
+    def set_user_karma(self, user_id, karma):
+        cursor = self.cursor()
+        cursor.execute(
+            (
+                'INSERT INTO KARMA_TRACKER VALUES (?, ?) '
+                'ON CONFLICT(user_id) DO UPDATE SET '
+                'karma=excluded.karma'
+            ),
+            (user_id, karma),
+        )
+        self.conn.commit()
+
+    def get_user_karma(self, user_id):
+        cursor = self.cursor()
+        karma = cursor.execute(
+            'SELECT karma FROM KARMA_TRACKER WHERE user_id=:user_id',
+            {'user_id': user_id},
+        ).fetchone()[0]
```

Comment:
pretty sure [`fetchone()` will return `None`](https://docs.python.org/3/library/sqlite3.html?highlight=fetchone#sqlite3.Cursor.fetchone) and will throw an error when you index with `[0]`. At least that's what the documentation makes me think. Did you test this without setting a user's karma?

Reply:
ack

Reply:
ack, now it doesn't produce any errors if the returned rows is None